src/main/java/us/kbase/auth/client/internal/StringCache.java

```diff
@@ -0,0 +1,139 @@
+package us.kbase.auth.client.internal;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Caches strings with an expiry time.
+ * 
+ * Strings are stored until the size of the cache is greater than the maximum
+ * allowed size. Strings are then ordered by most recent access and the oldest
+ * strings are discarded to return the cache to its nominal size.
+ * 
+ * This class is thread safe.
+ * 
+ * @author gaprice@lbl.gov
+ *
+ */
+public class StringCache {
+	// is there some way to share code with tokencache? Doesn't seem so...
+	
+	// TODO CODE replace this with Caffeine.
+	
+	/**
+	 * Default expiry time for strings.
+	 */
+	final public static long EXPIRY = 24 * 60 * 60;
```

Comment:
Not sure why we need this `EXPIRY` here as we already got getter and setter for private expiry.

Reply:
It's the default value if it's not set by the setter

Reply:
This just allows users to see the default value

Reply:
does that mean if we had setter and getter for MAX_AGE_MS in TokenCache, then no need for reflection in test?

Reply:
yes, but we don't want people changing the length of the token cache because it's used everywhere. The string cache is only used in certain places so we can be more lenient